src/components/ReportingConfig/index.jsx

```diff
@@ -0,0 +1,175 @@
+import ReportingConfigForm from './ReportingConfigForm';
+import { camelCaseObject, snakeCaseFormData } from '../../utils';
+import LoadingMessage from '../LoadingMessage';
+import ErrorPage from '../ErrorPage';
+
+class ReportingConfig extends React.Component {
+  state = {
+    loading: true,
+    reportingConfigs: [],
+    error: undefined,
+  };
+
+  // eslint-disable-next-line react/sort-comp
+  newConfigFormRef = React.createRef();
+
+  componentDidMount() {
+    LMSApiService.fetchReportingConfigs(this.props.enterpriseId)
+      .then(response => this.setState({
+        reportingConfigs: response.data.results,
+        loading: false,
+      }))
+      .catch(error => this.setState({ error }));
+  }
+
+  /**
+   * Creates a new reporting configuration, then updates this list with the response.
+   * Returns if there is an error.
+   * @param {FormData} formData
+   */
+  createConfig = async (formData) => {
+    //  snake_case the data before sending it to the backend
+    const transformedData = snakeCaseFormData(formData);
+    try {
+      const response = await LMSApiService.postNewReportingConfig(transformedData);
+      this.setState(prevState => ({
+        reportingConfigs: [
+          ...prevState.reportingConfigs,
+          response.data,
+        ],
+      }));
+      this.newConfigFormRef.current.close();
+      return undefined;
+    } catch (error) {
+      return error;
+    }
+  }
+
+  deleteConfig = async (uuid) => {
+    try {
+      const response = await LMSApiService.deleteReportingConfig(uuid);
+      const deletedIndex = this.state.reportingConfigs
+        .findIndex(config => config.uuid === response.data.uuid);
+      const reportingConfigsCopy = this.state.reportingConfigs;
+      reportingConfigsCopy.splice(deletedIndex, 1);
+      this.setState({
+        reportingConfigs: reportingConfigsCopy,
+      });
+      return undefined;
+    } catch (error) {
+      return error;
+    }
+  }
+
+  /**
+   * Updates an existing reporting configuration. Returns if there is an
+   * error.
+   * @param {FormData} formData
+   * @param {string} uuid -> The uuid of the configuration we are updating
+   */
+  updateConfig = async (formData, uuid) => {
+    const transformedData = snakeCaseFormData(formData);
+    try {
+      const response = await LMSApiService.updateReportingConfig(transformedData, uuid);
+      const updatedIndex = this.state.reportingConfigs
+        .findIndex(config => config.uuid === response.data.uuid);
+      const reportingConfigsCopy = [...this.state.reportingConfigs];
+      reportingConfigsCopy[updatedIndex] = response.data;
+      this.setState({ reportingConfigs: [...reportingConfigsCopy] });
+      return undefined;
+    } catch (error) {
+      return error;
+    }
+  }
+
+  render() {
+    const { reportingConfigs, loading, error } = this.state;
+
+    if (loading) {
+      return <LoadingMessage className="overview" />;
+    }
+
+    if (error) {
+      return <ErrorPage />;
```

Comment:
I think we'll want to add some status / message props to the ErrorPage

Reply:
done.